### DISCUSSION TITLE
Broader search options for public-facing docs

### DISCUSSION THREAD

**Discussion by sofietoft** [GitbookIO/community#18] [Feature Requests]
### What's your feature/idea?

We document multiple different products, each following a specific version cadence.
Currently, we have everything in the collection and space, which allows us to search for everything all at once.
However, we are planning on splitting it up, so that each product will get its own space/collection within the Organization.

Being able to search and also filter the search based on all collections and spaces within our Organization would be a great asset for our public-facing documentation.

### What's your desired outcome?

Being able to search everything everywhere all at once (and using filters).

### What's the impact of this for you?

- [ ] Critical
- [x] High
- [ ] Medium
- [ ] Low

**Comment by taszap** [comment]
Thanks for sharing this!

Concept of broader search options (or cross-space search on public docs) is something we have been discussing, as it's a feedback we hear quite often! This does require a more advanced research from our side before we can commit to it, but it's something we would like to do in the future. We will keep you posted on our progress here!

**Reply by sofietoft** [reply]
That's great to hear 🙌 

It would definitely be a sweet feature to offer our users/readers as many of our products are usually used in unison!

**Reply by jumptheknot** [reply]
I would also like the ability to search within a draft. The current search field within GitBook only allows you to search content in published spaces.

**Comment by kristenoneill** [comment]
We are running into the same issue - we are building 2 collections - one or our end user documentation and one for our developer docs - there is a ton of crossover that is lost when restricting the search to just one collection

**Comment by dfinster** [comment]
+1 for us as well. 

I posed this question in the #gitbook channel at Write the Docs Slack and got a nice reply from @addisonschultz asking for more details.

We'd like to publish one collection with multiple spaces. For a hypothetical example, consider a library aligned with the [Diátaxis framework](https://diataxis.fr/). In that case, a customer should have the ability to search across four spaces: tutorials, how-to guides, technical reference, and explanation. 

We'd like:
* Ability to see results from all spaces in one query.
* Results tagged with the space name, so customers know the content category of each result.
* Ability to filter by space, so they can narrow the results to a particular space. 

This is a high priority for us. Without this ability, we may need to explore other publishing tools, or consider how to use a service like Algolia instead of GitBook native search.

**Reply by addisonschultz** [reply]
Definitely some great use-cases, thanks for the context! We're aware this part of the product could be improved. We're actually working on an integration platform that will let you tap into GitBook's search API and customize the way you're looking for information. You can already find some documentation on that API here: https://developer.gitbook.com/gitbook-api/reference/search

I've also made a quick API explorer, if you were interested in checking out what information/data is returned! It searches across all spaces you're a part of (there is the ability to specify spaces to search from, but not in this demo/explorer app unfortunately). 

https://api-explorer.gitbook.dev/

You'll need to create a personal access token, which you can find here: https://app.gitbook.com/account/developer

Let me know if you have any questions on this, and if it's helpful at all!

**Reply by ghost** [reply]
@addisonschultz this looks interesting, and seems to work well. Looks like the "only search what you have access to" issue is a non-issue. :)

Who do we nag, pester, and otherwise harass to get the idea tested and released?

**Reply by addisonschultz** [reply]
We've still been collecting all feedback on this, but still don't have a clear timeline/roadmap on this feature unfortunately!

**Reply by ghost** [reply]
@addisonschultz what are the chances of getting this capability added as an integration? I am frantically trying to unforget all the TypeScript I've not used since the early 10's, but I'm not sure I can even begin to create anything suitable in the time frame I've been given.

An integration might help @sofietoft, @kristenoneill, and the rest of us without having to wait for code changes.

**Reply by addisonschultz** [reply]
I don't know if we have a lot of bandwidth in the team right now to tackle something like this, unfortunately.. It's definitely high on the list though for us to explore once we do have a bit more room to work on some integrations!

**Reply by BrainstemStudio** [reply]
I would like to suggest adding the ability to API search from specified spaces. This would allow us to curate and API search multiple spaces for our products that make use of various components which are documented in individual spaces. Like other users here, we also decided to split up our docs into spaces to make it easier for users to read in isolation, provide breathing room to get more detailed, and not have to repeat documentation for each product.

For example, adding in an optional spaces parameter for API search seems to make the most sense. Our current workaround is to use Google Search Console but this is less ideal and would rather use something more direct.

**Comment by volkantash** [comment]
<details>
<summary>Read 🇬🇧 English 👈👇</summary>

With Gitbook, we presented all our documents to the visitors as clicked and accessible on a common page.

When you click the search box and do a search, no results come up. This gives the visitor the idea that we don't have any content.

I really don't understand what you're arguing about what you think.

1- We will enter the search settings of the document.
2- We will choose which documents will participate in the search operations.
3- There may be several pages in each document that we do not want to appear in the search results. We will close the document pages we do not want.
4- Pray, we do not want to close the paragraphs that we do not want to appear in the search results. But why not this? 🤭

Although bringing the search feature with artificial intelligence makes you happy, there is nothing that makes us users happy. Who has eyes for chocolate when we are hungry? Forgive me for writing in a somewhat strange language. Thanks.
</details>

<details open>
<summary>🇹🇷Türkçe oku 👈👇</summary>
Gitbook ile bütün belgelerimizi ortak bir sayfada tıklanıp erişilebilir olarak ziyaretçiye sunduk. 

Arama kutusuna tıklayıp bir arama yaptığınızda hiçbir sonuç gelmiyor. Bu da ziyaretçiye hiçbir içeriğimizin olmadığı düşüncesini veriyor.

Gerçekten neyi düşündüğünüzü tartıştığınızı anlayamıyorum.

1- Belgenin arama ayarlarına gireceğiz.
2- Hangi belgelerin arama işlemlerine katılacağını seçeceğiz.
3- Arama sonuçlarında çıkmasını istemediğimiz birkaç sayfa her belgede olabilir. İstemediğimiz belge sayfalarını kapayacağız.
4- Dua edin, arama sonuçlarında çıkmasını istemediğimiz paragrafları kapamayı istemiyoruz. Ama bu da niye olmasın? 🤭

Yapayzeka ile arama özelliği getirmek sizi mutlu etse de biz kullanıcıları mutlu eden bir şey yok. Karnımız aç iken çikolatada kimin gözü olur? Biraz tuhaf bir dil ile yazıştığım için bağışlayın. Sağ olun.
</details>

**Comment by seanthomasdevlin** [comment]
Is there any update on this feature request?

If we cannot search multiple spaces, then we have to put everything in a single space, which kind of defeats the purpose of having spaces.

I would think that the default behavior should be to search all spaces within a collection.

**Comment by dmacros** [comment]
Same problem here. We have different products in our organization and they all have a lot of user manual content and tutorials. This is why we need to publish the information of each product in different spaces, for a better user experience.

Now we want to generate an initial support screen that has a multi-space search field. That is why we need this functionality.

**Comment by ismaen** [comment]
+1 on this request as well. We're considering moving to gitbook for our docs, but right now our docs provider does provide this functionality which is one of the features we are looking for. Everything else that gitbook does is fantastic! Would love to see this implemented!

**Comment by AndreiSerghiutaGit** [comment]
+1 from me as well. We also have a collection with spaces, and most of our users are confused when the search box does not find articles as expected. It would be great to allow the search to be performed across all spaces in a collection. Additional features to indicate the space where each result is located would be the cherry on top.

**Comment by ghost** [comment]
We are migrating from an internal system to gitbook. Our internal system was divided into areas of responsibility by repository. Each repository basically behaved like the concept of a space.

To "regain" our ability to search all the repositories will need to be combined into a single "space", which is going to make for one screaming lulu of a `MANIFEST.md`, that will be very unpleasant to navigate.

Having the ability to search across spaces would absolutely make our lives easier, and improve our experience.

**Comment by thebiglabasky** [comment]
Hi everyone,
We've been releasing an entirely revamped published content experience which now supports searching across all spaces published under the site.
Let us know what you think!
